Fix for corrupted CO tag issue

lbergelson opened the pull request:
* Someone on the forum reported an issue with a bam tag that was a long string instead of 2 characters.  This is a fix, but not a test because i'm not sure how to build a bam that triggers isContaminant.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

pshapiro4broad commented on src/main/java/picard/sam/AbstractAlignmentMerger.java:
```diff
@@ -694,8 +694,7 @@ private void transferAlignmentInfoToFragment(final SAMRecord unaligned, final SA
             Optional<String> optionalComment = Optional.ofNullable(unaligned.getStringAttribute(SAMTag.CO.name()));
-            unaligned.setAttribute(optionalComment.map(s -> s + " | ").orElse("") +
-                    SAMTag.CO.name(), "Cross-species contamination");
+            unaligned.setAttribute( SAMTag.CO.name(), optionalComment.map(s -> s + " | ").orElse("")  + "Cross-species contamination");
```
The fix looks good to me. It's a shame the API isn't `setAttribute(SAMTag, Object)`.
```suggestion
            unaligned.setAttribute(SAMTag.CO.name(), optionalComment.map(s -> s + " | ").orElse("")  + "Cross-species contamination");
```

lbergelson replied:
There's a pr open to do what you suggested.

lbergelson commented:
@pshapiro4broad I updated it to master and now it's failing with:

```
Gradle suite > Gradle test > picard.vcf.UpdateVcfSequenceDictionaryTest > testUpdateVcfSequenceDictionary[2](/dev/stdout) FAILED
    java.lang.IllegalArgumentException: Output format type is not set, or could not be inferred from the output path. If a path was used, does it have a valid VCF extension (.vcf, .vcf.gz, .bcf)?
        at htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder.build(VariantContextWriterBuilder.java:462)
        at htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder.build(VariantContextWriterBuilder.java:415)
        at picard.vcf.UpdateVcfSequenceDictionary.doWork(UpdateVcfSequenceDictionary.java:89)
        at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:305)
        at picard.vcf.UpdateVcfSequenceDictionaryTest.testUpdateVcfSequenceDictionary(UpdateVcfSequenceDictionaryTest.java:105)
Finished 4700 tests
```

Not sure what that's about.  Rebased it again to be sure and pushing, maybe it's just something transient.

lbergelson commented:
Seems to be fixed.  Will merged once this all passes.
